RecoEgamma/EgammaIsolationAlgos/plugins/EgammaTowerIsolationProducer.cc

```diff
@@ -21,65 +21,52 @@
 class EgammaTowerIsolationProducer : public edm::stream::EDProducer<> {
 public:
   explicit EgammaTowerIsolationProducer(const edm::ParameterSet&);
-  ~EgammaTowerIsolationProducer() override;
 
   void produce(edm::Event&, const edm::EventSetup&) override;
 
 private:
   // ----------member data ---------------------------
 
-  edm::InputTag emObjectProducer_;
-  edm::InputTag towerProducer_;
+  const edm::EDGetTokenT<edm::View<reco::Candidate>> emObjectProducer_;
+  const edm::EDGetTokenT<CaloTowerCollection> towerProducer_;
 
   double egHcalIsoPtMin_;
   double egHcalIsoConeSizeOut_;
   double egHcalIsoConeSizeIn_;
   signed int egHcalDepth_;
-
-  edm::ParameterSet conf_;
 };
 
 #include "FWCore/Framework/interface/MakerMacros.h"
 DEFINE_FWK_MODULE(EgammaTowerIsolationProducer);
 
-EgammaTowerIsolationProducer::EgammaTowerIsolationProducer(const edm::ParameterSet& config) : conf_(config) {
-  // use configuration file to setup input/output collection names
-  emObjectProducer_ = conf_.getParameter<edm::InputTag>("emObjectProducer");
-
-  towerProducer_ = conf_.getParameter<edm::InputTag>("towerProducer");
+EgammaTowerIsolationProducer::EgammaTowerIsolationProducer(const edm::ParameterSet& config)
+    : emObjectProducer_{consumes(config.getParameter<edm::InputTag>("emObjectProducer"))}
 
```

Comment:
curious how clang-format left this extra whitespace here, which looks rather odd

Reply:
That happens, unfortunately. I'm fixing it manually.